pkg/model/bootstrapscript.go

```diff
@@ -124,7 +139,19 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup, cs *kops.Cluste
 			spec["taints"] = ig.Spec.Taints
 			hooks := b.getRelevantHooks(ig.Spec.Hooks, ig.Spec.Role)
 			if len(hooks) > 0 {
-				spec["hooks"] = hooks
+				if hooksFingerprint, err := b.fingerprintSpecList(hooks); err != nil {
```

Comment:
hmmm ... thinking out loud, perhaps it's overkill; I know we loose the diff (i.e. content) but it might be useful to keep some context, rather then a large hash on the full spec, maybe a hash per item. At least them you have some clue as to what changed. @KashifSaadat @justinsb

Reply:
I like that idea, will give it a go!

Reply:
I've tried this out as suggested